cherrypicks.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/google/go-github/v28/github"
+	"github.com/sirupsen/logrus"
+
+	tgithub "github.com/mendersoftware/integration-test-runner/client/github"
+)
+
+// suggestCherryPicks suggests cherry-picks to release branches if the PR has been merged to master
+func suggestCherryPicks(log *logrus.Entry, pr *github.PullRequestEvent, githubClient tgithub.Client, conf *config) error {
+	// ignore PRs if they are not closed and merged
+	action := pr.GetAction()
+	if action != "closed" || !(*pr.PullRequest.Merged) {
+		log.Infof("Ignoring cherry-pick suggestions for action: %s, merged: %v", action, *pr.PullRequest.Merged)
+		return nil
+	}
+
+	// ignore PRs if they don't target the master branch
+	baseRef := *pr.PullRequest.Base.Ref
+	if baseRef != "master" {
+		log.Infof("Ignoring cherry-pick suggestions for base ref: %s", baseRef)
+		return nil
+	}
+
+	// initialize the git work area
+	repo := *pr.Repo.Name
+	tmpdir, err := ioutil.TempDir("", repo)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	gitcmd := exec.Command("git", "init", ".")
+	gitcmd.Dir = tmpdir
+	out, err := gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "github", "git@github.com:mendersoftware/"+repo)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// fetch the master branch
+	gitcmd = exec.Command("git", "fetch", "github", "master:local")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// fetch the PR branch
+	prNumber := strconv.Itoa(pr.GetNumber())
+	prBranchName := "pr_" + prNumber
+	gitcmd = exec.Command("git", "fetch", "github", "pull/"+prNumber+"/head:"+prBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// count the number of non-none changelog entries
+	baseSHA := *pr.PullRequest.Base.SHA
+	countCmd := exec.Command("sh", "-c", "git log "+baseSHA+"...pr_"+prNumber+" | grep -i \"Changelog: \" | grep -v -i \": none\" | wc -l")
```

Comment:
Suggest being a bit more explicit, since it's not uncommon to include the word "changelog" in various other contexts for example when doing releases, and to have the word "none" in an actual changelog message:

```
grep -i \"^Changelog:\" | grep -v -i \"^Changelog: *none\"
```

Might be useful to have a test for these sentences too. Examples would be:

* ```
  Fix changelog: This was actually reverted.

  Cancel-changelog: abcdef0123456789
   ```
* ```
  Check for nil.

  Changelog: Fix API returning none.
```

Reply:
Nice, thanks! Your grep machinery is way better than mine, I applied it.